File: android/app/src/main/java/com/rafiq/almomen/widgets/PrayerWidgetProvider.java
package com.rafiq.almomen.widgets;

import android.app.PendingIntent;
import android.appwidget.AppWidgetManager;
import android.appwidget.AppWidgetProvider;
import android.content.Context;
import android.content.Intent;
import android.widget.RemoteViews;
import com.rafiq.almomen.MainActivity;
import com.rafiq.almomen.R;

public class PrayerWidgetProvider extends AppWidgetProvider {

    @Override
    public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
        for (int appWidgetId : appWidgetIds) {
            updateAppWidget(context, appWidgetManager, appWidgetId);
        }
    }

    @Override
    public void onEnabled(Context context) {
    }

    @Override
    public void onDisabled(Context context) {
    }

    static void updateAppWidget(Context context, AppWidgetManager appWidgetManager, int appWidgetId) {
        RemoteViews views = new RemoteViews(context.getPackageName(), R.layout.widget_prayer);
        
        Intent intent = new Intent(context, MainActivity.class);
        PendingIntent pendingIntent = PendingIntent.getActivity(
            context,
            0,
            intent,
            PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE
        );
        
        views.setOnClickPendingIntent(R.id.widget_main_layout, pendingIntent);

        views.setTextViewText(R.id.text_next_prayer_name, "صلاة العصر");
        views.setTextViewText(R.id.text_countdown, "بقي 01:20:00");
        views.setTextViewText(R.id.text_date, "14 رمضان 1446");

        appWidgetManager.updateAppWidget(appWidgetId, views);
    }
}

File: android/app/src/main/java/com/rafiq/almomen/widgets/PrayerWidgetProvider.kt
package com.rafiq.almomen.widgets

import android.app.PendingIntent
import android.appwidget.AppWidgetManager
import android.appwidget.AppWidgetProvider
import android.content.Context
import android.content.Intent
import android.widget.RemoteViews
import com.rafiq.almomen.MainActivity
import com.rafiq.almomen.R

class PrayerWidgetProvider : AppWidgetProvider() {

    override fun onUpdate(
        context: Context,
        appWidgetManager: AppWidgetManager,
        appWidgetIds: IntArray
    ) {
        // There may be multiple widgets active, so update all of them
        for (appWidgetId in appWidgetIds) {
            updateAppWidget(context, appWidgetManager, appWidgetId)
        }
    }

    override fun onEnabled(context: Context) {
        // Enter relevant functionality for when the first widget is created
    }

    override fun onDisabled(context: Context) {
        // Enter relevant functionality for when the last widget is disabled
    }

    companion object {
        internal fun updateAppWidget(
            context: Context,
            appWidgetManager: AppWidgetManager,
            appWidgetId: Int
        ) {
            // Construct the RemoteViews object
            val views = RemoteViews(context.packageName, R.layout.widget_prayer)
            
            // Create an Intent to launch MainActivity
            val intent = Intent(context, MainActivity::class.java)
            val pendingIntent = PendingIntent.getActivity(
                context,
                0,
                intent,
                PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
            )
            
            views.setOnClickPendingIntent(R.id.widget_main_layout, pendingIntent)

            // Setup temporary data (In a real app, calculate actual offline prayer times here or load from SharedPreferences)
            views.setTextViewText(R.id.text_next_prayer_name, "صلاة العصر")
            views.setTextViewText(R.id.text_countdown, "بقي 01:20:00")
            views.setTextViewText(R.id.text_athkar, "«سبحان الله وبحمده»")
            views.setTextViewText(R.id.text_hijri_date, "14 رمضان 1446")

            // Instruct the widget manager to update the widget
            appWidgetManager.updateAppWidget(appWidgetId, views)
        }
    }
}
